src/main/java/uk/gov/mca/beacons/service/registrations/GetAllBeaconsService.java

```diff
@@ -0,0 +1,138 @@
+import javax.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.mca.beacons.service.model.Beacon;
+import uk.gov.mca.beacons.service.model.BeaconPerson;
+import uk.gov.mca.beacons.service.model.BeaconUse;
+import uk.gov.mca.beacons.service.model.PersonType;
+import uk.gov.mca.beacons.service.repository.BeaconPersonRepository;
+import uk.gov.mca.beacons.service.repository.BeaconRepository;
+import uk.gov.mca.beacons.service.repository.BeaconUseRepository;
+
+@Service
+@Transactional
+public class GetAllBeaconsService {
+
+  private final BeaconRepository beaconRepository;
+  private final BeaconUseRepository beaconUseRepository;
+  private final BeaconPersonRepository beaconPersonRepository;
+
+  @Autowired
+  public GetAllBeaconsService(
+    BeaconRepository beaconRepository,
+    BeaconUseRepository beaconUseRepository,
+    BeaconPersonRepository beaconPersonRepository
+  ) {
+    this.beaconRepository = beaconRepository;
+    this.beaconPersonRepository = beaconPersonRepository;
+    this.beaconUseRepository = beaconUseRepository;
+  }
+
+  public List<Beacon> findAll() {
+    final List<Beacon> allBeacons = beaconRepository.findAll();
+    if (allBeacons.size() == 0) return emptyList();
+
+    final Map<UUID, List<BeaconUse>> usesGroupedByBeaconId = getAllUsesGroupedByBeaconId();
+    final Map<PersonType, List<BeaconPerson>> personsGroupedByType = getAllPersonsGroupedByType();
+    final Map<UUID, BeaconPerson> ownersGroupedByBeaconId = getAllOwnersGroupedByBeaconId(
+      personsGroupedByType
+    );
+    final Map<UUID, List<BeaconPerson>> emergencyContactsGroupedByBeaconId = getAllContactsGroupedByBeaconId(
+      personsGroupedByType
+    );
+
+    return mapBeaconRelationships(
+      allBeacons,
+      usesGroupedByBeaconId,
+      ownersGroupedByBeaconId,
+      emergencyContactsGroupedByBeaconId
+    );
+  }
+
+  private List<Beacon> mapBeaconRelationships(
+    final List<Beacon> allBeacons,
+    final Map<UUID, List<BeaconUse>> usesGroupedByBeaconId,
+    final Map<UUID, BeaconPerson> ownersGroupedByBeaconId,
+    final Map<UUID, List<BeaconPerson>> emergencyContactsGroupedByBeaconId
+  ) {
+    List<Beacon> mappedBeacons = new ArrayList<Beacon>();
+
+    allBeacons.forEach(
+      beacon -> {
+        var beaconUses = usesGroupedByBeaconId.get(beacon.getId());
+        beacon.setUses(beaconUses);
+        var beaconOwner = ownersGroupedByBeaconId.get(beacon.getId());
+        beacon.setOwner(beaconOwner);
+        beacon.setEmergencyContacts(
+          emergencyContactsGroupedByBeaconId.get(beacon.getId())
+        );
+        mappedBeacons.add(beacon);
+      }
+    );
+
+    return mappedBeacons;
+  }
+
+  private Map<UUID, List<BeaconUse>> getAllUsesGroupedByBeaconId() {
+    final var usesStream = StreamSupport.stream(
+      beaconUseRepository.findAll().spliterator(),
+      false
+    );
```

Comment:
@tm1303 You can remove the `StreamSupport` class now as the `findAll` method will return a `List` of entries